js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -861,21 +878,53 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         $icon.hide();
     },
 
-    _renderToggleItemVisibilityIcon: function($node, node) {
-        const $icon = $('<div>')
-            .addClass(TOGGLE_ITEM_VISIBILITY_CLASS)
-            .appendTo($node);
+    _renderToggleItemVisibilityCustomIcon: function($node, node) {
+        const $expandIcon = getImageContainer(this.option('expandButtonIcon') ?? this.option('collapseButtonIcon'));
+        const $collapseIcon = getImageContainer(this.option('collapseButtonIcon') ?? this.option('expandButtonIcon'));
+
+        $expandIcon.appendTo($node).addClass('dx-custom-expander-icon');
+        $collapseIcon.appendTo($node).addClass('dx-custom-expander-icon');
+
+        $expandIcon.addClass(CUSTOM_ICON_TOGGLE_ITEM_VISIBILITY_CLASS);
+        $expandIcon.addClass(CUSTOM_EXPAND_ICON_CLASS);
+
+        $collapseIcon.addClass(CUSTOM_ICON_TOGGLE_ITEM_VISIBILITY_CLASS);
+        $collapseIcon.addClass(CUSTOM_COLLAPSE_ICON_CLASS);
+
+        this._toggleCustomExpanderBtnIcons($expandIcon, $collapseIcon, node.internalFields.expanded);
 
         if(node.internalFields.expanded) {
-            $icon.addClass(TOGGLE_ITEM_VISIBILITY_OPENED_CLASS);
```

Comment:
it works a little different, i switch visibility, but here we add class that helps to set content with :before pseudoclass. I hoped we will rewrite this and use my approach, then we set visibility instead of switching content with :before